File: app/src/main/java/com/example/a39092/rxjavatest/model/netWork/ThreadUtils.java
package com.example.a39092.rxjavatest.model.netWork;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by 39092 on 2017/4/5.
 */

class ThreadUtils {
    static <T> ObservableTransformer<T, T> IO_MAIN() {
        return new ObservableTransformer<T, T>() {
            @Override
            public ObservableSource<T> apply(Observable<T> upstream) {
                return upstream.subscribeOn(Schedulers.io()).unsubscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread());
            }
        };
    }
}

File: app/src/main/java/com/example/a39092/rxjavatest/model/netWork/ThreadUtils.kt
package com.example.a39092.rxjavatest.model.netWork

import io.reactivex.Observable
import io.reactivex.ObservableSource
import io.reactivex.ObservableTransformer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * Created by 39092 on 2017/4/5.
 */

internal object ThreadUtils {
    fun <T> IO_MAIN(): ObservableTransformer<T, T> {
        return ObservableTransformer { upstream -> upstream.subscribeOn(Schedulers.io()).unsubscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()) }
    }
}
